src/os/os_freertos.c

```diff
@@ -0,0 +1,121 @@
+#include "../types.h"
+#include "../oscore.h"
+#include "../main.h"
+#include "../timers.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/semphr.h"
+#include <sys/time.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdlib.h>
+
+#define STACK_DEPTH 16384
+#define TASK_PRIORITY 10
+#define N_CPUCORES 2
+
+static void user_main(void* pvParameters) {
+	sled_main(0, NULL);
+}
+
+// Main entry point.
+// We can choose to pin to a core. This makes sense if your chip has more than 1 core.
+// So you can pin SLED to one core and use the other cores for stuff like WiFi, Webserver, etc...
+// Use core = -1 to not pin
+void sled_task(void* pvParameters, int core) {
+	if(xTaskGetSchedulerState() == 1)
+		vTaskStartScheduler();
+	if (core < 0 || core >= oscore_ncpus()) {
+		xTaskCreate(user_main, "sled_main", STACK_DEPTH, pvParameters, TASK_PRIORITY, NULL);
+	} else {
+		xTaskCreatePinnedToCore(user_main, "sled_main", STACK_DEPTH, pvParameters, TASK_PRIORITY, NULL, core);
```

Comment:
If you could rename the the sled_task function to user_main (or whatever your entry point is) and also rename the user_main function to sled_task, that'd be nice.
They are still confusing.